app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,66 @@
+module CourseSummariesHelper
+  #
+  #  Get JSON data for the table in the form of
+  #  { id: student_id, user_name: student.user_name,
+  #    first_name: student.first_name, last_name: student.last_name
+  #    marks: [ // an array of marks for each assignment, null if no grade ] }
+  #
+  def get_table_json_data
+    all_students = Student.where(type: 'Student')
+    all_assignments = Assignment.all
+
+    student_list = all_students.map do |student|
+      {
+      id: student.id,
+      user_name: student.user_name,
+      first_name: student.first_name,
+      last_name: student.last_name,
+      marks: \
+        get_mark_for_all_assignments_for_student(student, all_assignments)
+      }
+    end
+      student_list.to_json
+  end
+
+  # Get marks for all assignments for a student
+  def get_mark_for_all_assignments_for_student(student, all_assignments)
+    marks = []
+    all_assignments.each do |assignment|
+      marks[assignment.id - 1] = \
+      get_mark_for_assignment_and_student(assignment, student)
+    end
+    marks
+  end
+
+  # Get mark for a perticular assignment for a student
```

Comment:
The helper functions here and below really shouldn't be necessary. MarkUs already displays grades to individual students when they log in; you should find, and reuse, the existing functions which accomplish this, rather than making your own here.